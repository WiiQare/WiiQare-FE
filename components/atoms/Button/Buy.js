import { Fragment, useState } from 'react'
import { HiOutlinePlus } from "react-icons/hi";
import { Dialog, Transition } from '@headlessui/react'
import ContentModal from '../Modal/content';

const ButtonBuy = ({ withIcon = true, modal, className, children }) => {
    let [isOpen, setIsOpen] = useState(false)

    const closeModal = () => {
        setIsOpen(false)
    }

    const openModal = () => {
        setIsOpen(true)
    }

    return (
        <>
            <button
                onClick={openModal}
                htmlFor="my-modal-6" className={`flex items-center gap-2 justify-between rounded-lg bg-orange py-2 px-3 effect-up shadow ${className}`} >
                {withIcon ? (
                    <span className='bg-white text-sky p-1 rounded-md'>
                        <HiOutlinePlus />
                    </span>
                ) : <></>}
                <span className='text-sm font-light'>Buy now</span>
            </button>

            <Transition appear show={isOpen} as={Fragment}>
                <Dialog as="div" className="relative z-50" onClose={closeModal}>
                    <Transition.Child
                        as={Fragment}
                        enter="ease-out duration-300"
                        enterFrom="opacity-0"
                        enterTo="opacity-100"
                        leave="ease-in duration-200"
                        leaveFrom="opacity-100"
                        leaveTo="opacity-0"
                    >
                        <div className="fixed inset-0 bg-black bg-opacity-25" />
                    </Transition.Child>
                    <ContentModal title={modal.title}>
                        {children}
                    </ContentModal>
                </Dialog>
            </Transition>
        </>
    );
}

export default ButtonBuy;
