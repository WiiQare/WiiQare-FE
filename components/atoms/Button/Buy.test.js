import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Buy from "./Buy";

describe("Buy button", () => {
  it("should render the component", () => {
    const { container } = render(<Buy title="Sample Text" />);
    expect(container).toMatchSnapshot();
  });

  it("should render the component", () => {
    render(<Buy />);
    expect(screen.getByRole("link")).toBeInTheDocument();
  });

  it("should render the text", () => {
    const text = "Button";
    render(<Buy title={text} />);
    expect(screen.getByText(text)).toBeInTheDocument();
  });

  it("should render the iconw without icon", () => {
    render(<Buy withIcon={false} />);
    expect(screen.getByRole("link")).toBeInTheDocument();
  });
});
