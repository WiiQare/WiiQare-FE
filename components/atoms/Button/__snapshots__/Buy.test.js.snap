// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy button should render the component 1`] = `
<div>
  <a
    class="flex items-center gap-2 justify-between rounded-lg bg-orange py-2 px-3 effect-up shadow undefined"
    href="#"
  >
    <span
      class="bg-white text-sky p-1 rounded-md"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4v16m8-8H4"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </span>
    <span
      class="text-sm font-light"
    >
      Sample Text
    </span>
  </a>
</div>
`;
